src/docfx/lib/log/SourceInfo.cs

```diff
@@ -0,0 +1,26 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Diagnostics;
+
+namespace Microsoft.Docs.Build
+{
+    [DebuggerDisplay("{Value}")]
+    public class SourceInfo<T> : ISourceInfo
+    {
+        public readonly T Value;
+
+        public readonly string File;
+
+        public readonly Range Range;
```

Comment:
can we add a `implicit` operator here, so we don't need to change most of the interface？

Reply:
I am thinking about turning `Range` into `SourceInfo` eventually, because a line number alone cannot identify the true source. (need both line info + file info)

Reply:
i mean the `value`

Reply:
There was once a weird bug due to implicit conversion of `JValue` https://github.com/dotnet/docfx/blob/v3/src/docfx/cli/CommandLineOptions.cs#L23 when it is used with `? :` and other operators, type inference may produce unexpected result. It used to be `a ? a : null` but was changed to `a ? (JValue)a : JValue.CreateNull()` 

Let's see how it goes for this one